test/letters/actions/letters.unit.spec.js

```diff
@@ -21,6 +36,208 @@ const frontEndAddress = {
   state: 'secret'
 };
 
+// TO-DO: Determine if this is needed
+const addressResponse = {
+  data: {
+    attributes: {
+      address: {
+        type: 'DOMESTIC',
+        addressEffectiveDate: '1973-01-01T05:00:00.000+00:00',
+        addressOne: '140 Rock Creek Church Rd NW',
+        addressTwo: '',
+        addressThree: '',
+        city: 'Washington',
+        stateCode: 'DC',
+        zipCode: '20011',
+        zipSuffix: '1865'
+      },
+      controlInformation: {
+        canUpdate: true,
+        corpAvailIndicator: true,
+        corpRecFoundIndicator: true,
+        hasNoBdnPaymentsIndicator: true,
+        isCompetentIndicator: true,
+        indentityIndicator: true,
+        indexIndicator: true,
+        noFiduciaryAssignedIndicator: true,
+        notDeceasedIndicator: true
+      }
+    }
+  }
+};
+
+/**
+ * Setup() for each test requires stubbing global fetch() and setting userToken
+ * in global sessionStorage. Teardown() resets everything back to normal
+ */
+let oldFetch;
+let oldSessionStorage;
+const setup = () => {
+  oldSessionStorage = global.sessionStorage;
+  oldFetch = global.fetch;
+  global.sessionStorage = {
+    userToken: '123abc'
+  };
+  global.fetch = sinon.stub();
+  global.fetch.returns(Promise.resolve({
+    headers: { get: () => 'application/json' },
+    ok: true,
+    json: () => Promise.resolve({})
+  }));
+};
+const teardown = () => {
+  global.fetch = oldFetch;
+  global.sessionStorage = oldSessionStorage;
+};
+const getState = () => ({});
+
+// without redux mock store - these tests pass
+describe('saveAddress', () => {
+  beforeEach(setup);
+  afterEach(teardown);
+
+  it('dispatches SAVE_ADDRESS_PENDING first', (done) => {
+    const thunk = saveAddress(frontEndAddress);
+    let callCount = 0;
+    const dispatch = sinon.spy((action) => {
+      // keep track of which dispatch() call we're asserting against
+      callCount += 1;
+      const { type } = action;
+      if (callCount === 1) {
+        try {
+          expect(type).to.equal(SAVE_ADDRESS_PENDING);
+          done();
+        } catch (error) {
+          done(error);
+        }
+      }
+    });
+
+    thunk(dispatch, getState);
+  });
+
+  it('dispatches SAVE_ADDRESS_SUCCESS on update success', (done) => {
+    const thunk = saveAddress(frontEndAddress);
+    const dispatch = sinon.spy((action) => {
+      const { type, address } = action;
+      if (type === SAVE_ADDRESS_SUCCESS) {
+        try {
+          expect(type).to.equal(SAVE_ADDRESS_SUCCESS);
```

Comment:
At this point, we already know this is going to pass, so there's no need for this assertion. The trick here is to catch if `dispatch` isn't called with the success action.

🤔  I think it's fine because `done` is never called if the action is never `SAVE_ADDRESS_SUCCESS`, and the test will timeout. The error message won't be super useful, but it should be obvious after we look at the test, though.

Alright, after puzzling this out, I'm okay with this. 👍